src/index.test.tsx

```diff
@@ -0,0 +1,238 @@
+    price: 116,
+    count: 1,
+    color: 'silver',
+  },
+];
+
+describe('ctablex', () => {
+  it('should render table', () => {
+    const c = (
+      <DataTable data={data}>
+        <Columns>
+          <Column>
+            <IndexCell />
+          </Column>
+          <Column header="Name" accessor="name">
+            <StringCell />
+          </Column>
+          <Column header="Price" accessor="price">
+            <StringCell />
+          </Column>
+          <Column header="Count" accessor="price">
+            <StringCell />
+          </Column>
+          <Column header="Total" accessor={(r: any) => r.price * r.count}>
+            <StringCell />
+          </Column>
+          <Column header="Color" accessor="color">
+            <StringCell />
+          </Column>
+        </Columns>
+        <Table>
+          <TableHeader>
+            <HeaderRow />
+          </TableHeader>
+          <TableBody>
+            <Rows>
+              <Row />
+            </Rows>
+          </TableBody>
+        </Table>
+      </DataTable>
+    );
+    const { unmount } = render(c);
+    unmount();
+  });
+  it('should provide index context', () => {
+    const c = (
+      <DataTable data={data}>
+        <Columns>
+          <Column>
+            <IndexCell />
+          </Column>
+          <Column header="Name" accessor="name">
+            <StringCell />
+          </Column>
+        </Columns>
+        <Table>
+          <TableHeader>
+            <HeaderRow />
+          </TableHeader>
+          <TableBody>
+            <Rows>
+              <Row />
+            </Rows>
+          </TableBody>
+        </Table>
+      </DataTable>
+    );
+    const { unmount } = render(c);
+    unmount();
+  });
+  it('should use custom adapter', () => {
+    const adapter = {
+      table: (props: PropsWithChildren<{}>) => {
+        return (
+          <table style={{ backgroundColor: '#ccc' }}>{props.children}</table>
+        );
+      },
+    };
+
+    const c = (
+      <TableUIAdapterContextProvider value={adapter}>
+        <DataTable data={data}>
+          <Columns>
+            <Column header="Name" accessor="name">
+              <StringCell />
+            </Column>
+          </Columns>
+          <Table>
+            <TableHeader>
+              <HeaderRow />
+            </TableHeader>
+            <TableBody>
+              <Rows>
+                <Row />
+              </Rows>
+            </TableBody>
+          </Table>
+        </DataTable>
+      </TableUIAdapterContextProvider>
+    );
+    const { unmount } = render(c);
+    unmount();
+  });
+  it('should use custom key accessor', () => {
+    const c = (
+      <DataTable data={data}>
+        <Columns>
+          <Column header="Name" accessor="name">
+            <StringCell />
+          </Column>
+        </Columns>
+        <Table>
+          <TableHeader>
+            <HeaderRow />
+          </TableHeader>
+          <TableBody>
+            <Rows keyAccessor="id">
```

Comment:
I think you can pass a mocked function to `keyAccessor` and see how many times and with what values it has been called